---
title: "flagstats"
author: "Samuel DeMario"
date: "11/12/2021"
output: html_document
---

```{r setup, include=FALSE}
library(ggplot2)
library("ggpubr")
library(plotly)
```

```{r}
my.t.test.p.value <- function(c1, c2, c3, c4) {
    ttest <- c("error")
    for (i in 1:nrow(split_pairs)) {
      obj<-try(t.test(x=split_pairs[i,c1:c2],y=split_pairs[i,c3:c4]), silent=TRUE)
      if (is(obj, "try-error")) {ttest[i] <- (NA)} else {ttest[i] <- (obj$p.value)}
    }
    return(as.numeric(ttest))
}
```


```{r}
r1<-read.table(file = 'Flagstats/RRP6_1YP_1.fq.gz.tsv', sep = '\t', header = FALSE, row.names = 3)
r2<-read.table(file = 'Flagstats/RRP6_2YP_1.fq.gz.tsv', sep = '\t', header = FALSE, row.names = 3)
r3<-read.table(file = 'Flagstats/RRP6_3YP_1.fq.gz.tsv', sep = '\t', header = FALSE, row.names = 3)
UR1<-read.table(file = 'Flagstats/U_R_1YPD_1.fq.gz.tsv', sep = '\t', header = FALSE, row.names = 3)
UR2<-read.table(file = 'Flagstats/U_R_2YPD_1.fq.gz.tsv', sep = '\t', header = FALSE, row.names = 3)
UR3<-read.table(file = 'Flagstats/U_R_3YPD_1.fq.gz.tsv', sep = '\t', header = FALSE, row.names = 3)
W1<-read.table(file = 'Flagstats/WT_1_YPD_1.fq.gz.tsv', sep = '\t', header = FALSE, row.names = 3)
W2<-read.table(file = 'Flagstats/WT_2_YPD_1.fq.gz.tsv', sep = '\t', header = FALSE, row.names = 3)
W3<-read.table(file = 'Flagstats/WT_3_YPD_1.fq.gz.tsv', sep = '\t', header = FALSE, row.names = 3)
U1<-read.table(file = 'Flagstats/UPF1_1YP_1.fq.gz.tsv', sep = '\t', header = FALSE, row.names = 3)
U2<-read.table(file = 'Flagstats/UPF1_2YP_1.fq.gz.tsv', sep = '\t', header = FALSE, row.names = 3)
U3<-read.table(file = 'Flagstats/UPF1_3YP_1.fq.gz.tsv', sep = '\t', header = FALSE, row.names = 3)
```

```{r}
flagstats_df <- data.frame( total = c(r1[1,1],
                      r2[1,1],
                      r3[1,1],
                      UR1[1,1],
                      UR2[1,1],
                      UR3[1,1],
                      W1[1,1],
                      W2[1,1],
                      W3[1,1],
                      U1[1,1],
                      U2[1,1],
                      U3[1,1]),
            mate_mapped = c(r1[16,1],
                      r2[16,1],
                      r3[16,1],
                      UR1[16,1],
                      UR2[16,1],
                      UR3[16,1],
                      W1[16,1],
                      W2[16,1],
                      W3[16,1],
                      U1[16,1],
                      U2[16,1],
                      U3[16,1]),
            mutant = c("rrp6",
                       "rrp6",
                       "rrp6",
                       "upf1rrp6",
                       "upf1rrp6",
                       "upf1rrp6",
                       "WT",
                       "WT",
                       "WT",
                       "upf1",
                       "upf1",
                       "upf1"
                       ))

flagstats_df$total <- as.numeric(flagstats_df$total)
flagstats_df$mate_mapped <- as.numeric(flagstats_df$mate_mapped)
flagstats_df$mutant <- as.factor(flagstats_df$mutant)

flagstats_df$percent_flagged <- (flagstats_df$mate_mapped/flagstats_df$total)*100
```

```{r}
split_pairs <- read.table("./Count_tables/split_pairs.txt", header = TRUE)
ALL_reads <- read.table("./Count_tables/all_reads.txt", header = TRUE)

fixed_names<- c(
  "rrp6_1",
  "rrp6_2",
  "rrp6_3",
  "upf1_1",
  "upf1_2",
  "upf1_3",
  "upf1_rrp6_1",
  "upf1_rrp6_2",
  "upf1_rrp6_3",
  "WT_1",
  "WT_2",
  "WT_3")

colnames(split_pairs)[7:18] <- fixed_names
colnames(ALL_reads)[7:18] <- fixed_names

split_pairs$rrp6_mean <- rowMeans(split_pairs[7:9])
split_pairs$upf1_mean <- rowMeans(split_pairs[10:12])
split_pairs$upf1_rrp6_mean <- rowMeans(split_pairs[13:15])
split_pairs$WT_mean <- rowMeans(split_pairs[16:18])

split_pairs$rrp6_vs_upf1_rrp6 <- split_pairs$rrp6_mean/split_pairs$upf1_rrp6_mean

split_pairs$rrp6_vs_upf1_rrp6_t_test <- my.t.test.p.value(7,9,13,15)
split_pairs$wt_vs_upf1_t_test <- my.t.test.p.value(16,18,10,12)

ALL_reads$rrp6_mean <- rowMeans(ALL_reads[7:9])
ALL_reads$upf1_mean <- rowMeans(ALL_reads[10:12])
ALL_reads$upf1_rrp6_mean <- rowMeans(ALL_reads[13:15])
ALL_reads$WT_mean <- rowMeans(ALL_reads[16:18])

for (i in 1:nrow(split_pairs)) {
  split_pairs[i,"WT_over_split"] <- split_pairs[i,"WT_mean"]/ALL_reads[i,"WT_mean"]
  split_pairs[i,"upf1_over_split"] <- split_pairs[i,"upf1_mean"]/ALL_reads[i,"upf1_mean"]
}

for (i in 1:nrow(split_pairs)) {
  split_pairs[i,"WT1_over_split"] <- split_pairs[i,"WT_1"]/ALL_reads[i,"WT_mean"]
  split_pairs[i,"WT2_over_split"] <- split_pairs[i,"WT_2"]/ALL_reads[i,"WT_mean"]
  split_pairs[i,"WT3_over_split"] <- split_pairs[i,"WT_3"]/ALL_reads[i,"WT_mean"]
  split_pairs[i,"upf1_1_over_split"] <- split_pairs[i,"upf1_1"]/ALL_reads[i,"upf1_mean"]
  split_pairs[i,"upf1_2_over_split"] <- split_pairs[i,"upf1_2"]/ALL_reads[i,"upf1_mean"]
  split_pairs[i,"upf1_3_over_split"] <- split_pairs[i,"upf1_3"]/ALL_reads[i,"upf1_mean"]
}

split_pairs$wt_vs_upf1_ratio_t_test <- my.t.test.p.value(28,30,31,33)

high_abundance_split_pairs <- subset(split_pairs, split_pairs$WT_mean > 100)

high_unsplit_abundance_split_pairs <- subset(split_pairs, ALL_reads$WT_mean > 100000)

chr_split <- strsplit(split_pairs$Chr, split = ";")
for (i in 1:length(chr_split)) {
  split_pairs$intron[i] <- length(chr_split[[i]]) != 1
}

intron_split_pairs <- subset(split_pairs, split_pairs$intron)

intron_split_pairs_high_abundance <- subset(intron_split_pairs, intron_split_pairs$WT_mean > 100)

```

```{r}
plot_with_t_test <- function (df, title) {
  
  df_low_pval <- subset(df,df[,"wt_vs_upf1_ratio_t_test"]<0.05)
  
  wt_vs_upf1 <- ggplot(data = df)+
  geom_point(aes(x=WT_over_split, y=upf1_over_split, text=Geneid, color = wt_vs_upf1_ratio_t_test))+
  scale_color_gradient2(low = "#FF0000C1", mid = "#334CFFC1", high = "#000000C1", midpoint = -1.3, trans = "log10")+
  scale_x_log10()+
  scale_y_log10()+
  geom_abline(slope = 1)+
  labs(title = title)+
  xlab(label = "WT Split Reads / WT Total Reads")+
  ylab(label = "upf1d Split Reads / upf1d Total Reads")+
  geom_point(data = df_low_pval,aes(x=WT_over_split, y=upf1_over_split, text=Geneid), color = "Green")
  

  return(wt_vs_upf1)
}

allannotations <- plot_with_t_test(split_pairs,"All Annotations")
intron_wthigh <- plot_with_t_test(intron_split_pairs_high_abundance,"Intron containing (CDS) Annotations | WT mean > 100")
allannotations_wthigh <- plot_with_t_test(high_abundance_split_pairs,"All Annotations | WT split mean > 100")
introns <- plot_with_t_test(intron_split_pairs,"Intron containing (CDS) Annotations")
all_wtunsplithigh <- plot_with_t_test(high_unsplit_abundance_split_pairs,"All Annotations | WT unsplit > 10000")

ggsave(filename = "plots/11NOV2021/allannotations.png", plot = allannotations, device = 'png', units = "in",height = 4, width = 7)
ggsave(filename = "plots/11NOV2021/intron_wthigh.png", plot = intron_wthigh, device = 'png', units = "in",height = 4, width = 7)
ggsave(filename = "plots/11NOV2021/allannotations_wthigh.png", plot = allannotations_wthigh, device = 'png', units = "in",height = 4, width = 7)
ggsave(filename = "plots/11NOV2021/introns.png", plot = introns, device = 'png', units = "in",height = 4, width = 7)
ggsave(filename = "plots/11NOV2021/all_wtunsplithigh.png", plot = all_wtunsplithigh, device = 'png', units = "in",height = 4, width = 7)

# wt_vs_upf1_paired_intron_high_abundance <- ggplot(data = intron_split_pairs_high_abundance)+
#   geom_point(aes(x=WT_over_split, y=upf1_over_split, text=Geneid, color = wt_vs_upf1_ratio_t_test))+
#   scale_x_log10()+
#   scale_y_log10()+
#   geom_abline(slope = 1)+
#   labs(title = "Intron containing (CDS) Annotations | WT mean > 100")+
#   xlab(label = "WT Split Reads / WT Total Reads")+
#   ylab(label = "upf1d Split Reads / upf1d Total Reads")

# wt_vs_upf1_paired_high_abundance <- ggplot(data = high_abundance_split_pairs)+
#   geom_point(aes(x=WT_over_split, y=upf1_over_split, text=Geneid, color = wt_vs_upf1_ratio_t_test))+
#   scale_x_log10()+
#   scale_y_log10()+
#   geom_abline(slope = 1)+
#   labs(title = "All Annotations | WT split mean > 100")+
#   xlab(label = "WT Split Reads / WT Total Reads")+
#   ylab(label = "upf1d Split Reads / upf1d Total Reads")

# wt_vs_upf1_paired_introns <- ggplot(data = intron_split_pairs)+
#   geom_point(aes(x=WT_over_split, y=upf1_over_split, text=Geneid, color = wt_vs_upf1_ratio_t_test))+
#   scale_x_log10()+
#   scale_y_log10()+
#   geom_abline(slope = 1)+
#   labs(title = "Intron containing (CDS) Annotations")+
#   xlab(label = "WT Split Reads / WT Total Reads")+
#   ylab(label = "upf1d Split Reads / upf1d Total Reads")

# wt_vs_upf1_paired_unsplit_abundance <- ggplot(data = high_unsplit_abundance_split_pairs)+
#   geom_point(aes(x=WT_over_split, y=upf1_over_split, text=Geneid, color = wt_vs_upf1_ratio_t_test))+
#   scale_x_log10()+
#   scale_y_log10()+
#   geom_abline(slope = 1)+
#   labs(title = "All Annotations | WT unsplit > 10000")+
#   xlab(label = "WT Split Reads / WT Total Reads")+
#   ylab(label = "upf1d Split Reads / upf1d Total Reads")


ggplotly(wt_vs_upf1_paired_all)
ggplotly(wt_vs_upf1_paired_intron_high_abundance)
ggplotly(wt_vs_upf1_paired_high_abundance)
ggplotly(wt_vs_upf1_paired_introns)
ggplotly(wt_vs_upf1_paired_unsplit_abundance)
```

```{r}
my_comparisons <- list( c("WT", "upf1"))

box_plot <- ggplot(data = flagstats_df)+
  geom_boxplot(aes (x = mutant, y = percent_flagged, fill =mutant))+
  xlab("Mutant")+
  ylab("% of total reads mapped to diffrent chromosomes") +
  stat_compare_means(ref.group = "WT", aes(x = mutant, y = percent_flagged), method = "t.test")

ggsave(filename = "plots/11NOV2021/boxplot.png", plot = box_plot, device = 'png', units = "in",height = 4, width = 7)

```

```{r}
box_plot_all <- ggplot(data = split_pairs)+
  geom_boxplot(aes (x = 1, y = WT1_over_split),fill ="blue1")+
  geom_boxplot(aes (x = 2, y = WT2_over_split), fill ="blue2")+
  geom_boxplot(aes (x = 3, y = WT3_over_split), fill ="blue3")+
  geom_boxplot(aes (x = 4, y = WT_over_split), fill ="blue4")+
  geom_boxplot(aes (x = 5, y = upf1_1_over_split), fill ="red1")+
  geom_boxplot(aes (x = 6, y = upf1_2_over_split), fill ="red2")+
  geom_boxplot(aes (x = 7, y = upf1_3_over_split), fill ="red3")+
  geom_boxplot(aes (x = 8, y = upf1_over_split), fill ="red4")+
  scale_y_log10()+
  xlab("Mutant")+
  ylab("% of total reads mapped to diffrent chromosomes") 

box_plot_introns <- ggplot(data = intron_split_pairs)+
  geom_boxplot(aes (x = 1, y = WT1_over_split),fill ="blue1")+
  geom_boxplot(aes (x = 2, y = WT2_over_split), fill ="blue2")+
  geom_boxplot(aes (x = 3, y = WT3_over_split), fill ="blue3")+
  geom_boxplot(aes (x = 4, y = upf1_1_over_split), fill ="red1")+
  geom_boxplot(aes (x = 5, y = upf1_2_over_split), fill ="red2")+
  geom_boxplot(aes (x = 6, y = upf1_3_over_split), fill ="red3")+
  scale_y_log10()+
  xlab("Mutant")+
  ylab("% of total reads mapped to diffrent chromosomes") 

box_plot_highabundance <- ggplot(data = high_abundance_split_pairs)+
  geom_boxplot(aes (x = 1, y = WT1_over_split),fill ="blue1")+
  geom_boxplot(aes (x = 2, y = WT2_over_split), fill ="blue2")+
  geom_boxplot(aes (x = 3, y = WT3_over_split), fill ="blue3")+
  geom_boxplot(aes (x = 4, y = upf1_1_over_split), fill ="red1")+
  geom_boxplot(aes (x = 5, y = upf1_2_over_split), fill ="red2")+
  geom_boxplot(aes (x = 6, y = upf1_3_over_split), fill ="red3")+
  scale_y_log10()+
  xlab("Mutant")+
  ylab("% of total reads mapped to diffrent chromosomes") 

t.test(x = split_pairs$upf1_over_split, y = split_pairs$WT_over_split)

```

